### PR TITLE
Feature/bai 75 containerize app

### PR DESCRIPTION
**Pull request by at55425**
Run the app in its own container instead of relying on host Docker.

Uses img for builds.

Uses docker for registry login (img doesn't send complete authorisation headers), but "docker login" doesn't require the docker daemon to be running. It populates .docker/config.json that img then uses for pushing the built image into the registry.

### CODE REVIEW

**Review by a3957273 — CHANGES_REQUESTED** [review]

File: server/routes/v1/upload.ts [hunk 1]
```diff
@@ -176,7 +176,7 @@ export const postUpload = [
         binary: normalizeMulterFile(files.binary[0]),
         code: normalizeMulterFile(files.code[0]),
       })
-      .timeout(60000)
+      .timeout(60000*8)
```
[the remark this answers is not in the document]
Comment (at55425) [reply server/routes/v1/upload.ts]: Recent tests are 6 m for first build (before seldon image was pulled by the app) and then 4.3s for another build (admittedly with the same underlying model binary and code)

[the remark this answers is not in the document]
Comment (at55425) [reply server/routes/v1/upload.ts]: Having the image pulled by the app took 4m30s (1 run), resulting in first initial model image build->push to registry time of 2m 55.5s.

[the remark this answers is not in the document]
Comment (at55425) [reply server/routes/v1/upload.ts]: (subsequent build of *same model* via separate model upload was 4.4s)

**Review by a3957273 — APPROVED** [review]

**Review by a3957273 — CHANGES_REQUESTED** [review]
Unsure about this given `seccomp:unconfined`

File: .github/workflows/node.js.yml [hunk 2]
```diff
@@ -29,12 +29,6 @@ jobs:
...
 
-      # Install s2i
-      - run: |
-          wget https://github.com/openshift/source-to-image/releases/download/v1.3.1/source-to-image-v1.3.1-a5a77147-linux-amd64.tar.gz -P ~
-          tar -xvf ~/source-to-image-v1.3.1-a5a77147-linux-amd64.tar.gz -C $HOME/
-          echo "$HOME/" >> $GITHUB_PATH
-
       # Linting
       - run: npm run check-style
       - run: npm run lint
```
Comment (a3957273) [inline .github/workflows/node.js.yml:34]: You no longer need to run `npm run build` here, as it's built inside of docker-compose.

File: .github/workflows/node.js.yml [hunk 3]
```diff
@@ -29,12 +29,6 @@ jobs:
       # Install dependencies
       - run: npm ci
```
Comment (a3957273) [inline .github/workflows/node.js.yml:30]: You no longer need to install dependencies here, as they're installed inside the container.

**Review by a3957273 — CHANGES_REQUESTED** [review]

**Review by ARADDCC002 — APPROVED** [review]

**Review by a3957273 — APPROVED** [review]

**Comment by at55425** [comment]
Run app in container

**Pull request merged**